SCClassLibrary/Common/Collections/String.sc

```diff
@@ -130,6 +130,7 @@ String[char] : RawArray {
 	format { arg ... items; ^this.prFormat( items.collect(_.asString) ) }
 	prFormat { arg items; _String_Format ^this.primitiveFailed }
 	matchRegexp { arg string, start = 0, end; _String_Regexp ^this.primitiveFailed }
+	replaceRegexp { |regex, with| _String_ReplaceRegex ^this.primitiveFailed }
```

Comment:
Shoud these args maybe default to an empty string instead of a `nil` value?

Reply:
I'm not sure about that.
 
The `regex` argument needs to be supplied as it doesn't make sense to call this without some pattern.

The `with` argument could default to an empty string but then the method would best be described as `removeRegex` rather than `replaceRegex`. I don't think there is a clean default value here.

Reply:
I think a common trait in sclang is to fail silently instead of throwing exceptions where possible. But I also agree that default empty strings don't make much sense here. Maybe @telephon has a good idea here which is idiomatic?

Reply:
I would say that keeping all default values nil is the most idiomatic one in this case.